T3/Gui/Interaction/UserAction.cs

```diff
@@ -126,6 +126,9 @@ public class KeyboardBinding
         public bool NeedsWindowHover;
         public bool KeyPressOnly;
         public readonly KeyCombination Combination;
+        public float AutoRepeatTime = -1f;
+        private float PressTime = 0;
```

Comment:
I believe, this is not optimal: The KeyboardBinding class should be considered "immutable" without any state. It's basically a hotkey definition. As soon as you add state that depends on user interaction, things start to get messy.

I can see however, how this is a tricky case because the "state" is being held by ImGui.

Are you sure, you can't use any of the ImGui features likes...

```
    // uses provided repeat rate/delay. return a count, most often 0 or 1 but might be >1 if RepeatRate is small enough that DeltaTime > RepeatRate
    ImGui.GetKeyPressedAmount(ImGuiKey key, float repeat_delay, float rate);  
```